### PR TITLE
Add test and fix for preventing a type error for props

### LINKED ISSUE #81: TypeError in is_new method
For some objects/values the equality operator does not support special cases like comparing values with `None`. Which can result in the following error:

```
  File "[..]/collagraph.py", line 591, in update_dom_or_component
    if is_new(val, prev_props, key):
  File "[..]/collagraph.py", line 638, in is_new
    return val != other.get(key)
TypeError: bad argument type for built-in operation
```
The `val` argument was of type `PySide6.QtCore.Qt.ItemFlags`.

Possible solution:
```python
def is_new(val, other, key):
    other_value = other.get(key)
    if other_value is None and val is not None:
        return True

    return val != other_value
```

### PR DESCRIPTION
**Pull request by berendkleinhaneveld**
Fixes #81 .

### CODE REVIEW

File: collagraph/collagraph.py [hunk 1]
```diff
@@ -634,7 +634,11 @@ def key_to_event(key):
 
 
 def is_new(val, other, key):
-    return val != other.get(key)
+    other_value = other.get(key)
+    if other_value is None and val is not None:
```
Comment (Korijn) [inline collagraph/collagraph.py:638]: do you need to check this the other way around as well?

Comment (berendkleinhaneveld) [reply collagraph/collagraph.py:638]: I wanted to say 'no', but you actually have a good point! I'll update the test to also check for this.

**Comment by berendkleinhaneveld** [comment]
This PR now depends on the work in https://github.com/fork-tongue/observ/pull/84.

File: collagraph/collagraph.py [hunk 2]
```diff
@@ -634,7 +635,11 @@ def key_to_event(key):
 
 
 def is_new(val, other, key):
-    return val != other.get(key)
+    other_value = other.get(key)
+    if xor(other_value is None, val is None):
+        return True
+
+    return val != other_value
```
Comment (Korijn) [inline collagraph/collagraph.py:642]:
```suggestion
return xor(old_value is None, new_value is None) or old_value != new_value:
```

File: collagraph/collagraph.py [hunk 3]
```diff
@@ -634,7 +635,8 @@ def key_to_event(key):
 
 
 def is_new(val, other, key):
-    return val != other.get(key)
+    other_value = other.get(key)
```
Comment (Korijn) [inline collagraph/collagraph.py:638]:
```suggestion
    old_value = other.get(key)
```

Sorry :)

**Review by Korijn — APPROVED** [review]

**Pull request merged**